lib/galaxy/tools/__init__.py

```diff
@@ -520,7 +520,6 @@ def tool_shed_repository(self):
                                                             name=self.repository_name,
                                                             owner=self.repository_owner,
                                                             installed_changeset_revision=self.installed_changeset_revision,
-                                                            repository_id=self.repository_id,
```

Comment:
Not sure I understand this change.

Reply:
If you specify a repository_id we'll not be able to use ToolConfRepository items, which don't have an id. We should only provide the repository_id when working in the galaxy_install module.